schema/encoding/jsonschema/all_test.go

```diff
@@ -0,0 +1,324 @@
+	"testing"
+
+	"github.com/rs/rest-layer/schema"
+	"github.com/rs/rest-layer/schema/encoding/jsonschema"
+	"github.com/stretchr/testify/assert"
+)
+
+// encoderTestCase is used to test the Encoder.Encode() function.
+type encoderTestCase struct {
+	name           string
+	schema         schema.Schema
+	expect         string
+	customValidate func(t *testing.T, result []byte)
+}
+
+// Run runs the testCase according to your Go version. For Go >= 1.7, test cases are run in parallel using the Go 1.7
+// sub-test feature. For older versions of Go, the testCase name is simply logged before tests are run sequentially.
+func (tc *encoderTestCase) Run(t *testing.T) {
+	tc.run(t)
+}
+
+// test performs the actual test and is used by both implementations of run.
+func (tc *encoderTestCase) test(t *testing.T) {
+	b := new(bytes.Buffer)
+	enc := jsonschema.NewEncoder(b)
+	assert.NoError(t, enc.Encode(&tc.schema))
+
+	if tc.customValidate == nil {
+		assert.JSONEq(t, tc.expect, b.String())
+	} else {
+		tc.customValidate(t, b.Bytes())
+	}
+}
+
+func TestEncoder(t *testing.T) {
+	testCases := []encoderTestCase{
+		// readOnly is a custom extension to JSON Schema, also defined by the Swagger 2.0 Schema Object
+		// specification. See http://swagger.io/specification/#schemaObject.
+		{
+			name: "ReadOnly=true",
+			schema: schema.Schema{
+				Fields: schema.Fields{
+					"name": schema.Field{
+						ReadOnly:  true,
+						Validator: &schema.String{},
+					},
+				},
+			},
+			expect: `{
+				"type": "object",
+				"properties": {
+					"name": {
+						"type": "string",
+						"readOnly": true
+					}
+				}
+			}`,
+		},
+		{
+			name: `type(Default)=string`,
+			schema: schema.Schema{
+				Fields: schema.Fields{
+					"name": schema.Field{
+						Validator: &schema.String{},
+						Default:   "Luke",
+					},
+				},
+			},
+			expect: `{
+				"type": "object",
+				"properties": {
+					"name": {
+						"type": "string",
+						"default": "Luke"
+					}
+				}
+			}`,
+		},
+		// Currently we do not validate the type for Default.  Note that according to the JSON Schema
+		// Specification Section 6.2 (http://json-schema.org/latest/json-schema-validation.html#anchor101), the
+		// default value for a schema is not strictly required to apply to the schema.  It's worth noticing that
+		// the Swagger 2.0 variant of JSON Schema (http://swagger.io/specification/#schemaObject), does require
+		// the default value to match the schema.
+		{
+			name: `type(Default)=int`,
+			schema: schema.Schema{
+				Fields: schema.Fields{
+					"name": schema.Field{
+						Validator: &schema.String{},
+						Default:   24,
+					},
+				},
+			},
+			expect: `{
+				"type": "object",
+				"properties": {
+					"name": {
+						"type": "string",
+						"default": 24
+					}
+				}
+			}`,
+		},
+		{
+			name: `type(Default)=map[string]interface{}`,
+			schema: schema.Schema{
+				Fields: schema.Fields{
+					"name": schema.Field{
+						Default: map[string]interface{}{
```

Comment:
I will change it to be less odd to avoid confusion, probably by setting the default one level out.